app/src/main/java/android/app/lotus/view/statistics/evaluation/Evaluation.kt

```diff
@@ -1,20 +1,301 @@
+                                answers.add(selectedOption.value)
+                                selectedOption.value = ""
+                                currentQuestion = nextQuestion(currentQuestion, questions.size)
+                            },
+                            shape = RoundedCornerShape(25.dp),
+                            borderColor = Color(148, 0, 124),
+                            borderStrokeWidth = 6.dp,
+                            size = 25.dp
+                        )
+                    }
+                }
+            }
+
+            QuestionType.TEXT_INPUT -> {
+                BasicTextField(
+                    value = textState.value,
+                    onValueChange = { textState.value = it },
+                    singleLine = true,
+                    textStyle = TextStyle(fontSize = 18.sp),
+                    keyboardOptions = KeyboardOptions.Default.copy(
+                        imeAction = ImeAction.Done
+                    ),
+                    keyboardActions = KeyboardActions(
+                        onDone = {
+                            answers.add(textState.value.text)
+                            currentQuestion = nextQuestion(currentQuestion, questions.size)
+                        }
+                    ),
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .height(100.dp)
+                        .background(Color(245, 245, 245))
+                        .border(1.dp, Color.Gray)
+                        .padding(20.dp, 0.dp, 0.dp, 0.dp)
+                )
+            }
+        }
+        Spacer(modifier = Modifier.height(20.dp))
+        if (currentQuestion == questions.size - 1) {
+            Button(
+                onClick = {
+                    when (questionTypes[currentQuestion]) {
+                        QuestionType.YES_NO, QuestionType.MULTIPLE_CHOICE -> {
+                            answers.add(selectedOption.value)
+                            selectedOption.value = ""
+                        }
+
+                        QuestionType.TEXT_INPUT -> {
+                            answers.add(textState.value.text)
+                        }
+                    }
+                    //logic when the "Submit" button is clicked.
+                },
+                modifier = Modifier
+                    .height(60.dp)
+                    .width(175.dp)
+                    .align(alignment = Alignment.CenterHorizontally),
+                colors = ButtonDefaults.buttonColors(Color(148, 0, 124)),
+                shape = RoundedCornerShape(25.dp),
+            ) {
+                Text(text = "Submit",
+                    fontSize = 20.sp,
+                    fontWeight = FontWeight.SemiBold
+                )
+            }
+        }
+// Add a Back button
+        val isBackButtonVisible = currentQuestion > 0
```

Comment:
When you click on `Back` the answer doesnt get deleted from `answers` list, I would reccomend using a map instead of a list for answers in this case

`[Yes, 2, 2, 2, 2, 2, , 5, , 5, 4]`

Reply:
Map of: `key: questionName, value: answer`